packages/parser/src/analyze-scope.ts

```diff
@@ -214,7 +220,7 @@ class Referencer extends OriginalReferencer {
       this.visit(node.superTypeParameters);
     }
     if (node.implements) {
-      this.visit(node.implements);
+      node.implements.forEach(this.visit, this);
```

Comment:
this is actual issue, `implements` in classes is a array of nodes but `visit` expect that you provide node iteslf

i have to design a way to test visiting, scope analysis snapshots does not contain information about implements, its part of type "scope", and its not yet implemented 

see #19